File: listme_backend/src/main/java/com/listme/dto/UserDTO.java
package com.listme.dto;

import com.listme.model.Cargo;
import com.listme.model.Usuario;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class UserDTO {
    private Integer idUsuario;
    private String login;
    private String nome;
    private String email;
    private LocalDateTime dataCriacao;
    private Boolean ativo;
    private Cargo cargo;

    public UserDTO(Usuario usuario) {
        this.idUsuario = usuario.getIdUsuario();
        this.login = usuario.getLogin();
        this.nome = usuario.getNome();
        this.email = usuario.getEmail();
        this.dataCriacao = usuario.getDataCriacao();
        this.ativo = usuario.getAtivo();
        this.cargo = usuario.getCargo();
    }
}

File: listmeapp/app/src/main/java/com/example/listmeapp/data/model/UserDTO.kt
package com.example.listmeapp.data.model

data class UserDTO(
    val idUsuario: Long,
    val login: String,
    val nome: String,
    val email: String,
    val dataCriacao: String, // String para simplicidade, pode ser Instant/Date com TypeAdapter
    val ativo: Boolean,
    val cargo: String
)